pyanaconda/gnome_remote_destop.py

```diff
@@ -0,0 +1,226 @@
+                         "-out", GRD_RDP_CERT,
+                         "-keyout", GRD_RDP_CERT_KEY]
+                        )
+        # tell GNOME remote desktop to use these certificates
+        self._run_grdctl(["rdp", "set-tls-cert", GRD_RDP_CERT])
+        self._run_grdctl(["rdp", "set-tls-key", GRD_RDP_CERT_KEY])
+
+    def _set_rdp_username_and_password(self):
+        """Set the RDP username and password."""
+        self._run_grdctl(["rdp", "set-credentials", self.rdp_username, self.rdp_password])
+        # disable view only mode
+        self._run_grdctl(["rdp", "disable-view-only"])
+        # also actually tell GNOME remote desktop that we (obviously) want to use RDP
+        self._run_grdctl(["rdp", "enable"])
+
+    def _find_network_address(self):
+        """Find machine IP address, so we can show it to the user."""
+
+        # Network may be slow. Try for 5 seconds
+        tries = 5
+        while tries:
+            self.ip = network.get_first_ip_address()
+            if self.ip:
+                self.log.info(self.ip)
+                break
+            time.sleep(1)
+            tries -= 1
+
+        if not self.ip:
+            return
+
+        # FIXME: resolve this somehow,
+        # so it does not get stuck for 2 minutes in some VMs
+
+        if self.ip.find(':') != -1:
+            ipstr = "[%s]" % (self.ip,)
+        else:
+            ipstr = self.ip
+
+        try:
+            hinfo = socket.gethostbyaddr(self.ip)
+            self.log.info(hinfo)
+            if len(hinfo) == 3:
+                # Consider as coming from a valid DNS record only if single IP is returned
+                if len(hinfo[2]) == 1:
+                    self.name = hinfo[0]
+        except socket.herror as e:
+            log.debug("Exception caught trying to get host name of %s: %s", ipstr, e)
+
+    def _run_grdctl(self, argv):
+        """Run grdctl in the correct environment.
+
+        This is necessary, as grdctl requires $HOME to be pruned
+        or else the call might not have the desired effect.
+        """
+        # we always run GRD in --headless mode
+        base_argv = ["--headless"]
+        # extend the base argv by the caller provided arguments
+        combined_argv = base_argv + argv
+        # make sure HOME is set to /root or else settings might not be saved
+        execWithCapture("grdctl", combined_argv, env_prune=['HOME'])
+
+    def _open_grd_log_file(self):
+        # FIXME: redirect to journal ?
+        try:
+            fd = os.open(GRD_LOG_FILE, os.O_RDWR | os.O_CREAT)
+        except OSError as e:
+            sys.stderr.write("error opening %s: %s\n", (GRD_LOG_FILE, e))
+            fd = None
+
+        return fd
+
+    def _start_grd_process(self):
+        """Start the GNOME remote desktop process."""
+        try:
+            self.log.info("Starting GNOME remote desktop.")
+            global grd_process
+            grd_process = startProgram([GRD_BINARY_PATH, "--headless"],
```

Comment:
Any reason to not use the systemd service (`gnome-remote-desktop-headless.service`)?

Reply:
Is it available in some released gnome-remote-desktop package ?

I have seen it here: https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/merge_requests/215/diffs

And that looks like it has been merged, but I only see `/usr/lib/systemd/user/gnome-remote-desktop.service` in `gnome-remote-desktop-45.1-1` that I have installed on my Fedora 39 machine and latest Rawhide seems to have justa a rebuild of that & there is indeed no such file in the RPM: https://kojipkgs.fedoraproject.org/compose/40/Fedora-40-20240319.1/compose/Everything/x86_64/os/Packages/g/gnome-remote-desktop-45.1-4.fc40.x86_64.rpm

So maybe the MR introducing this is merged but not yet packaged ?

Reply:
It seems so yes, merged but not packaged. Seems to be in progress (https://src.fedoraproject.org/rpms/gnome-remote-desktop/pull-request/6) - I'll see what is left to do.

Reply:
OK, thanks! :) We can switch to that service later when it becomes available in the environment.

Reply:
Maybe the spam with `--headless` and the `gnome-remote-desktop-headless.service` are related ? Because I guess that if the GNOME remote desktop binary was launched by system then maybe it would capture **all** of its output & it will end up in Journal (which we would like) instead of on TTY1 ?

Thanks to @jstodola or possibly @rvykydal for the idea, not sure which right now. :)

Reply:
I've created a bug in Bugzilla, so we can track the missing service file. :)
https://bugzilla.redhat.com/show_bug.cgi?id=2270718